samples/MachineLearning/R/forecast.R

```diff
@@ -344,7 +345,17 @@ smart.forecast <- function(dataset1, dataset2) {
 
   for (i in 1:granularityAttributes_num) {
     # Prepare training and test data for given granularity attribute
-    granularityAttribute_data <- data[which(data$GranularityAttribute == granularityAttributes[i]), ]
+    if (is.na(granularityAttributes[i])) {
+      granularityAttribute_data <- data[which(is.na(data$GranularityAttribute)), ]
+    }
+    else {
+      granularityAttribute_data <- data[which(data$GranularityAttribute == granularityAttributes[i]), ]
```

Comment:
I'm not an expert at R, but won't data[<empty>, ] always bring one row? Which means the nrow below will always return at least 1? Since there's a blank row (the one after the comma).

Reply:
I will leave as-is; code is tested in separate repository.